### PR TITLE
Combat oriented mech designs need Head access, Head ID swipe, station level red or higher, or emagged to be printable

### PR DESCRIPTION
**Pull request by Chubbygummibear**
# Document the changes in your pull request

You know what the purpose of this is. It's to try and mitigate the Durand for no reason, the Phazon for "research purposes", the gygax "because i was bored that's what roboticists do!!!"

I know this is coming out on the coattails of the townhall in which they said "we shouldn't be using code to prevent powergaming", so that's a fat L on my part, but i put in too much work to stop and i don't think this is detrimentally interrupting.

# So how does this work?
Everyone with less than head access cannot see

- Durand/Gygax/Honk/Phazon pieces
- Equipment that is purely combat purposed. Hades carbine, Solaris Cannon, and the ammo as well
image
image
Emagged
image

# So how do I get the gamer mechs?
If you have Head access or higher on your ID: You can see the combat designs
**SILICONS ARE ALWAYS AUTHORIZED** (i hate borgs)
If you use a Head access ID or higher and swipe the console: **Everyone** can see the combat designs
If the station's threat level is Red or higher: Everyone can see the combat designs
If the exosuit fab gets emagged, it breaks the authorization system, but be warned that this does make it visibly broken. Also this doesn't PUT the designs in the machine. If you roundstart emag the exofab and scream when there's no combat parts, you deserve what comes next

# Demonstration except i forgot to show that Silicons are always authorized
https://user-images.githubusercontent.com/46236974/182282443-28e1f648-52b5-4593-857a-3d9cee2e47cb.mp4

# Wiki Documentation

Combat mech parts only show up for heads, silicons, if a head overrides the authorization check, if the station threat level is red or greater, or the exofab is emagged.

# Changelog

:cl:  
rscadd: Exofabs now check your access to see if combat designs should be made available to you
tweak: Combat mech and equipment designs are only available to heads, silicons, heads who override the authorization check, if the security level is red or greater, or the exofab is emagged.
rscadd: An emagged exofab is very noticeable through the ui, but not from sight
/:cl:

### CODE REVIEW

**Comment by Chubbygummibear** [comment]
god damn commits

**Comment by NCRLouTenant** [comment]
I honestly don't know how to feel about this. On one hand,  the mech problem is one that has been an issue for a while but on the other hand being limited to what you can do as roboticist to someone else's whims sucks., especially given how little consequence there is between rounds.  I do like the variables like alert level and the multiple ways to get access, though.  

I'm somewhere in the middle on this one, but of all the recent "powergaming" PRs this is the one I could get behind the easiest.

**Comment by Spacemanspark** [comment]
Roboticists have no reason to be randomly constructing these in the first place, and being that the result of them doing such will get them bwoinked it makes sense to block it off. 

Also this means I'll be wanting to strangle my robotics coworker 2% less often and that's always a good thing for my cholesterol

**Comment by RG4ORDR** [comment]
I don't like this.
It's an emag tax for traitor robos now, and it makes it harder for team antags like Revs to be bold and create a machine that security has an answer for as well. 
Why isn't there some consideration to actually making restricting exo-suit weapons?
Why not actually reduce the effectiveness of T3 and T4 parts inside the mech, that's where a lot of the meat comes from these days.
Maybe make Ion carbine more affordable or w/e need only base weapon tech so security won't get bullrushed by it?
Add in cooldowns for how fast a mech can actually melee someone so it instant an instant game over?
Maybe make mechs not work in the rebee?
Due to ynot making an I DED PR you can't run up to a mech and attach an tracking beacon to EMP it, revert it.

**Comment by Chubbygummibear** [comment]
i see people throwing around "emag tax" a lot, but do consider that the current state of affairs is that for 0 tc you can build a combat mech that more or less requires emps to counter, meanwhile you can arm it with missiles, lasers, shotgun, lmg, self repair equipment, and a fuckload of armor or wacky abilities.
the two cases i see are

A: 
you are not a traitor and you shouldn't be building a combat mech unless there's a need for it, so it remains locked until the station is in red alert or a head overrides the console
B:
you are a traitor and if **YOU DECIDE** you want to take the combat mech route, you buy an item that's also ghetto all access in addition to the ability to create a ridiculously powerful tool. no one is forcing you to use a mech as a traitor roboticist, that's your choice

revs did come up in discussion about this change and my argument is your target is heads anyway. if you kill just one you only need to take their id to unlock the parts, or if the station goes sec level red in response to the rev threat they will also become available. additionally you can also subvert ai or a borg to get them to print the parts since silicons have access.

in short i don't think "tc tax" is a strong argument when currently the death robots cost 0 tc, and there's many ways to still get access with an ounce of effort

**Comment by ToasterBiome** [comment]
not a map thing

**Review by JamieD1 — APPROVED** [review]
